src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -4565,9 +4565,9 @@ internal bool TryReadSqlValue(SqlBuffer value, SqlMetaDataPriv md, int length, T
 
         private bool TryReadSqlDateTime(SqlBuffer value, byte tdsType, int length, byte scale, TdsParserStateObject stateObj)
         {
-            byte[] datetimeBuffer = new byte[length];
+            Span<byte> datetimeBuffer = length <= 10 ? stackalloc byte[length] : new byte[length];
```

Comment:
I recommend making this a constant power of two and making the argument to `stackalloc` also a constant power of two, e.g.:

```cs
Span<byte> foo = ((uint)length /* note uint */ <= 16) ? stackalloc byte[16 /* note const */] : new byte[length];
```

Reply:
(Context: I didn't review any other part of this PR, only the `stackalloc` call due to it triggering one of our internal flags.)

Reply:
Done. The uint change is pretty clear but why power of 2 alignment? Since we don't know what else is on the current stack frame I wouldn't have thought there were any alignment guarantees around stackalloc.